fix_auth: use random when fixing secret_token(s)

kbrock opened the pull request:
`fix_auth` is the tool users use when they lose a `v2_key`.
After losing a `v2_key`, the passwords can't be retrieved and are thus invalid.
The option of `--invalid bogus` resets these passwords to a known value.
The user can then go into the ui and correct.

The `MiqDatabase` table is an exception here. `session_secret_token` and `csrf_secret_token` do not contain provider passwords but rather random strings. So the invalid option works a little differently for this table. If you specify invalid, it will populate with a random string and not the `bogus` value provided by the user.

Unfortunately #962 introduces a bug where `session_secret_token` and `csrf_secret_token` are populated with `bogus` value.

This PR re-introduces the `hardcode` method that was overridden by `FixMiqDatabase`.

/cc @jrafanie @abellotti

**UPDATED:** reworded. thanks @abellotti 
**UPDATED:** The bug luckily was found before going out with 5.3 or 5.4

jrafanie commented on vmdb/tools/fix_auth/models.rb:
```diff
@@ -17,8 +17,18 @@ class FixMiqDatabase < ActiveRecord::Base
     self.password_columns = %w(registration_http_proxy_server session_secret_token csrf_secret_token)
 
-    def self.hardcode(old_value, _new_value)
-      super(old_value, SecureRandom.hex(64))
```
I can't find what `super` is doing.  Link to this method?  I only see this existing implementation.  
`vmdb/tools/fix_auth/models.rb:    def self.hardcode(old_value, _new_value)`

kbrock replied:
Hmm. Think re-introducing this will fix the bug as well. Think this will read better

since it was so simple, it was merged into `recrypt:
```ruby
def hardcode(_old_value, new_value)
  MiqPassword.encrypt(new_value)
end
```

jrafanie commented:
Minor nit:  commit message:
..Unfortunately it had a regressing with MiqDatabase and the hardcode...

Should be:
..Unfortunately it had a regression with MiqDatabase and the hardcode...

jrafanie commented on vmdb/tools/fix_auth/auth_model.rb:
```diff
@@ -25,15 +25,19 @@ def selection_criteria(include_v2 = false)
       rescue
         if options[:invalid]
-          MiqPassword.encrypt(options[:invalid])
+          hardcode(old_value, options[:invalid])
```
I don't understand this, it looks like the method was extracted but with the same behavior. Can you help me understand what this changes?

kbrock commented:
@jrafanie good question:

`AuthModel#hardcode` was [removed](https://github.com/ManageIQ/manageiq/pull/962/files#diff-496829dcccd434135864dffbc3357523L28). Which looks simple enough.

`MiqDatabase#hardcode` overrides `AuthModel#hardcode` [here](https://github.com/kbrock/manageiq/blob/79a63af92d82bd90f1b4acd6ed9d7aa091423e33/vmdb/tools/fix_auth/models.rb#L20). So removing that extension point means we no longer are calling into the overridden `hardcode` and not calling the correct logic.

My first fix overrode all of `AuthModel#recrypt` in `MiqDatabase`, but as you can see here, all that was needed was to reintroduce the `hardcode` method/extension point.

jrafanie commented:
Thanks @kbrock, I was :confused: but now I get it and the change looks good.

kbrock commented:
Thanks @jrafanie Was surprised that you didn't see the fix right away. Since it was basically spurred by something you had said while looking at it.

Appreciate the: "So what. You extracted a method. How does that fix anything?"
It makes me feel better because I had though the same when I removed that method and introduced the bug. :)

jrafanie commented:
Yeah, if it's not obvious, it's usually worthwhile asking.  :smiley_cat:
